webapp/src/errorList.tsx

```diff
@@ -94,7 +101,7 @@ export class ErrorList extends React.Component<ErrorListProps, ErrorListState> {
 
     onErrorsChanged(errors: pxtc.KsDiagnostic[]) {
         this.setState({
-            errors,
+            errors: groupErrors(errors),
```

Comment:
IMO, we should lazily group errors **if** the error list is visible. In most cases, it is collapsed and this computation is not required.

Reply:
> IMO, we should lazily group errors **if** the error list is visible. In most cases, it is collapsed and this computation is not required.

Agreed that's better. Updated.